packages/neuron-ui/src/components/GeneralSetting/index.tsx

```diff
@@ -108,40 +118,50 @@ const GeneralSetting = ({ updater }: GeneralSettingProps) => {
   useEffect(() => {
     const checkUpdate = searchParams.get('checkUpdate')
     if (checkUpdate === '1') {
-      setShowCheck(true)
+      checkForUpdates()
     }
-  }, [searchParams, setShowCheck])
+  }, [searchParams, checkForUpdates])
 
   useEffect(() => {
-    if (showCheck) {
-      setShowUpdateDialog(true)
-      if (updater.checking) {
-        setDialogType('checking')
-        return
-      }
-      if (updater.version || updater.downloadProgress > 0) {
-        setDialogType('updating')
-        return
-      }
+    if (updater.errorMsg) {
```

Comment:
How about adding `notify` when `cancelCheckUpdates` or `cancelDownloadUpdate`, then dialogType can be calculated by `updater` simply.